vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -124,6 +147,31 @@ def _set_slider_values(self):
             if self.selector.max is None:
                 self.selector.max = max(max_values)
 
+    def _set_date_picker_values(self):
+        if isinstance(self.selector, SELECTORS["temporal"]):
+            if self._column_type != "temporal":
+                raise ValueError(
+                    f"Chosen selector {self.selector.type} is not compatible "
+                    f"with {self._column_type} column '{self.column}'."
+                )
+
+            min_values = []
+            max_values = []
+            for target_id in self.targets:
+                data_frame = data_manager._get_component_data(target_id)
+                data_frame = self._convert_column_type(data_frame=data_frame)
+                min_values.append(data_frame[self.column].min())
+                max_values.append(data_frame[self.column].max())
+            if not is_datetime64_any_dtype(pd.Series(min_values)) or not is_datetime64_any_dtype(pd.Series(max_values)):
+                raise ValueError(
+                    f"Non-temporal values detected in the shared data column '{self.column}' for targeted charts. "
+                    f"Please ensure that the data column contains the same data type across all targeted charts."
+                )
```

Comment:
Do we actually need this? I remember we have this for the range slider, but which case does this actually catch?
Because you already convert the relevant data column before right?